SQL Store.adoc

```diff
@@ -0,0 +1,262 @@
+= SQL Store
+
+As a application developer or data grid administrator, I want to preload a cache from a relational database accessed by JDBC - via configuration without writing boilerplate code.
+
+== Architecture
+
+* the store can be either read/write or read-only (in which case it behaves as a loader)
+* the store can work in either single-table mode or in query-mode (via user supplied queries)
+* the store can work with "primitive" keys/values or with complex types defined as protobuf schemas
+* a protobuf schema can either be generated dynamically from the resultset metadata or supplied by the user
```

Comment:
AFAIK protostream currently only allows proto files and marshallers to be generated from Java objects, not the other way round. Is this something we plan to add to ProtoStream? Or will we simply use the google libs here?

Reply:
Generating a protobuf schema requires nothing more than string concatenation. No other libs required.

Reply:
I was thinking of the `application/x-java-object` use-case and forgetting that the loaded key/values are either `String` or `byte[]`.